src/ansys/fluent/core/session_solver.py

```diff
@@ -150,7 +150,9 @@ def _root(self):
         """Root settings object."""
         if self._settings_root is None:
             self._settings_root = settings_get_root(
-                flproxy=self._settings_service, version=self.version
+                flproxy=self._settings_service,
```

Comment:
Your architecture is ready to allow alternative file services to be injected. It's  a simple matter of requiring an object with upload and download methods. We could allow it to be inject at the pyfluent level statically such that it's simply picked up by each session object. That would be follow-up work.

Reply:
What is the objective of tracking upload and download requests other than unit testing ? because `assert len(file_service.uploads) == 0` will be the case always for local session and in the GitHub.

Reply:
tracking upload and download requests is what the mock service would do in the test

Reply:
I just asked this doubt because if we pass mock_instance to remote_file_handler in test then it returns remote_file_handler as a `NoneType` object. We can do upload/download using mock instance hence we can not track uploads/downloads, therefore we are using temporary `purpose` argument in the Ansys Lab.

Reply:
Your file service should have a simple interface such that alternative file services can easily be substituted. Right? If that alternative is a mock, you can implement it how you like. The mock will only be used in unit tests.

Reply:
Yes, I agree. Thank you.